### ISSUE TITLE
SNS Feedback role is broken

### ISSUE THREAD

**Issue by kvendingoldo** [terraform-aws-modules/terraform-aws-notify-slack#219]
## Description

The same as https://github.com/terraform-aws-modules/terraform-aws-notify-slack/issues/204

- [x] ✋ I have searched the open/closed issues and my issue is not listed.

## ⚠️ Note

Before you submit an issue, please perform the following first:

1. Remove the local `.terraform` directory (! ONLY if state is stored remotely, which hopefully you are following that best practice!): `rm -rf .terraform/`
2. Re-initialize the project root to pull down modules: `terraform init`
3. Re-attempt your terraform plan or apply and check if the issue still persists

## Versions

- Module version [Required]:

- Terraform version:

- Provider version(s):

**Comment by SwarajBaral** [comment]
Hey so I am working on a fix for this and wanted to verify if I am headed in the right direction before I raise PR. Apologies if this is not appropriate. 
I was trying to follow https://github.com/terraform-aws-modules/terraform-aws-notify-slack/issues/204#issuecomment-1733617233 and came up with this inline policy. Is this the right way to go about it ?
```terraform
  assume_role_policy = jsonencode({
    Version = "2012-10-17",
    Statement = [{
      Effect    = "Allow",
      Principal = {
        Service = "sns.amazonaws.com"
      },
      Action    = "sts:AssumeRole"
    }]
  })
```

**Comment by kclinden** [comment]
> Hey so I am working on a fix for this and wanted to verify if I am headed in the right direction before I raise PR. Apologies if this is not appropriate. I was trying to follow [#204 (comment)](https://github.com/terraform-aws-modules/terraform-aws-notify-slack/issues/204#issuecomment-1733617233) and came up with this inline policy. This will ensure that the IAM policy for CloudWatch Logs is properly attached to the role, and a valid assume role policy is defined, granting AWS SNS the right to assume the role. Is this the right way to go about it ?
> 
> ```terraform
>   assume_role_policy = jsonencode({
>     Version = "2012-10-17",
>     Statement = [{
>       Effect    = "Allow",
>       Principal = {
>         Service = "sns.amazonaws.com"
>       },
>       Action    = "sts:AssumeRole"
>     }]
>   })
> ```

This is the correct fix. You will also want to move the previous policy over to an inline policy document or create the policy as a standalone resource.

**Comment by antonbabenko** [comment]
This issue has been resolved in version 6.1.2 :tada:
